weave/ops_arrow/dict.py

```diff
@@ -111,6 +111,47 @@ def pick(self, key):
     return ArrowWeaveList(result, object_type, self._artifact)
 
 
+def _ensure_awl_for_merge(
+    ensure_target: typing.Union[dict, ArrowWeaveList],
+    compare_target: typing.Union[dict, ArrowWeaveList],
+) -> ArrowWeaveList:
+    if not isinstance(ensure_target, ArrowWeaveList):
+        assert isinstance(compare_target, ArrowWeaveList)
+        utility_awl = convert.to_arrow([ensure_target])
+        return ArrowWeaveList(
+            pa.repeat(ensure_target, len(compare_target)),
+            utility_awl.object_type,
+            utility_awl._artifact,
+        )
+    return ensure_target
+
+
+MaybeTypedDictType = types.optional(types.TypedDict({}))
+AWLMaybeTypedDictType = ArrowWeaveListType(MaybeTypedDictType)
+
+
+@op(
+    name="ArrowWeaveList-_preprocessMerge",
```

Comment:
do we use this underscore convention elsewhere? Is this just to make it extra hidden?

Reply:
exactly. its just to dissuade anyone from ever trying to use it